vexim/siteaddsubmit.php

```diff
@@ -62,10 +62,21 @@
     die;
   }
   if(isset($_POST['maildir']) && isset($_POST['localpart'])) {
-    $smtphomepath = realpath($_POST['maildir'] . "/") .
-      "/" . $_POST['domain'] . "/" . $_POST['localpart'] . "/Maildir";
-    $pophomepath = realpath($_POST['maildir'] . "/") .
-      "/" . $_POST['domain'] . "/" . $_POST['localpart'];
+    if (substr($_POST['maildir'], 0, 1) !== '/') {
+      header ("Location: site.php?failmaildirnonabsolute={$_POST['maildir']}");
+	  die();
+    }
+    if ($testmailroot && is_dir(realpath($_POST['maildir'])) === false) {
```

Comment:
If $testmailroot==0, you don't want to check the realpath. But if $testmailroot==0 this line will always be false because 
`FALSE && FALSE === FALSE`, `FALSE && TRUE === FALSE`. It should be
````
if ($testmailroot===1) { 
   if (is_dir(realpath($_POST['maildir']))===false) {
      ...
````
Or shorter:
````
if(!$testmailroot || is_dir(realpath($_POST['maildir'])) === false)
````
or:
````
if($testmailroot && !is_dir(realpath($_POST['maildir'])) === true)
````